pkg/clients/aws/s3/s3.go

```diff
@@ -181,10 +181,18 @@ func isErrorNotFound(err error) bool {
 
 // CreateBucketInput returns a CreateBucketInput from the supplied S3BucketSpec.
 func CreateBucketInput(spec *v1alpha1.S3BucketSpec) *s3.CreateBucketInput {
+	const (
+		// https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region
+		regionWithNoConstraint = "us-east-1"
+	)
 	bucketInput := &s3.CreateBucketInput{
-		CreateBucketConfiguration: &s3.CreateBucketConfiguration{LocationConstraint: s3.BucketLocationConstraint(spec.Region)},
-		Bucket:                    &spec.Name,
+		Bucket: &spec.Name,
 	}
+
+	if spec.Region != regionWithNoConstraint {
+		bucketInput.CreateBucketConfiguration = &s3.CreateBucketConfiguration{LocationConstraint: s3.BucketLocationConstraint(spec.Region)}
```

Comment:
this doesn't regress provisioning to regions other than us-east-1, does it?

Reply:
No I made the code purposefully obtuse so no one "optimizes" it by mistake later. Basically the code is doing what it's always done in other regions, but removing the locationConstraint for us-east-1.

Notice: https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region
"None required" next to us-east-1 actually means I'll barf on the api if you send me one.